### PR TITLE
Add Rancher Fleet page

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Relates to https://github.com/kubewarden/policy-server/issues/394.

Needs https://github.com/kubewarden/automation/pull/41

Explain how to use Rancher Fleet Bundles to codify Kubewarden dependencies.

### CODE REVIEW

File: docs/operator-manual/Rancher-Fleet.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+sidebar_label: "Rancher Fleet"
+title: ""
+---
+
+# Managing Kubewarden with Rancher Fleet
+
+The Kubewarden Helm charts, as other Helm charts, can be managed via [Rancher
```
Comment (divya-mohan0209) [inline docs/operator-manual/Rancher-Fleet.md:8]:
```suggestion
The Kubewarden Helm charts, like other Helm charts, can be managed via [Rancher
```

File: docs/operator-manual/Rancher-Fleet.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Fleet](https://fleet.rancher.io/). Rancher Fleet uses Kubernetes CRDs to define
+a GitOps approach for managing Kubernetes clusters. It does this by defining
+Fleet Bundles.
+
+The Kubewarden charts are normal charts, and they have defined dependencies that
+need to be ready before the dependent chart is deployed:
```
Comment (divya-mohan0209) [inline docs/operator-manual/Rancher-Fleet.md:14]: Do we want to interlink to existing documentation for each of these points?

File: docs/operator-manual/Rancher-Fleet.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- First, install `cert-manager`, as it is needed for the Kubewarden stack default config.
+- Second, install `kubewarden-crds`, as it is needed for the rest of Kubewarden charts.
+- Third, install `kubewarden-controller`, as it is needed by `kubewarden-defaults`.
+- Fourth, install `kubewarden-defaults`.
+
+On Rancher Fleet, one can codify the chart dependencies using [`dependsOn` in the _fleet.yaml_ file](https://fleet.rancher.io/gitrepo-structure#fleetyaml).
```
Comment (divya-mohan0209) [inline docs/operator-manual/Rancher-Fleet.md:20]:
```suggestion
On Rancher Fleet, one can codify the chart dependencies using `dependsOn` in the [_fleet.yaml_ file](https://fleet.rancher.io/gitrepo-structure#fleetyaml).
```

**Review by divya-mohan0209 — COMMENTED** [review]
Left a few suggestions, hope you find them useful!

File: docs/operator-manual/Rancher-Fleet.md (hunk 2 above)
Comment (flavio) [reply docs/operator-manual/Rancher-Fleet.md:16]: I personally think that would be a bit too much, I would however put a link to the quickstart installation docs

File: docs/operator-manual/Rancher-Fleet.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+finished deploying, they will be gone.
+
+### Removing
+
+Uninstalling CRDs automatically is normally not supported in any tooling, and
+Rancher Fleet is no exception.
+
+If you want to perform a correct removal, make sure to remove first the Bundle
+for `kubewarden-defaults` from the cluster, then `kubewarden-controller`, and
+last, `kubewarden-crds`.
```
Comment (flavio) [inline docs/operator-manual/Rancher-Fleet.md:41]: Can we provide some copy and paste docs about that?

File: docs/operator-manual/Rancher-Fleet.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+deletes the default policy-server. This pre-delete hook is needed because we
+need to vacate the webhooks of the policies (this is true any Policy Engine)
+before deleting the PolicyServer, or the cluster will have webhooks for
+policies, but no policies instatiated, and will reject everything and be in a
+failed state.
+
+Removing the GitRepo and hence the `kubewarden-crds` chart at the same time as
+the `kubewarden-controller` chart will make the pre-delete hook job to fail, and
+the removal to be incomplete, leaving leftovers in the cluster.
+
```
Comment (flavio) [inline docs/operator-manual/Rancher-Fleet.md:56]: I would wrap this block into a [docusaurus admonition](https://docusaurus.io/docs/markdown-features/admonitions), like a `caution`. This is dangerous stuff

File: docs/operator-manual/Rancher-Fleet.md (hunk 4 above)
Comment (viccuad) [reply docs/operator-manual/Rancher-Fleet.md:41]: It's not so easy. I can explain it by deleting the [`defaullts` bundle folder](https://github.com/viccuad/kubewarden-fleet/tree/main/helm-kubewarden/defaults) from the repo, committing it, waiting for Fleet to apply, removing the `controller` bundle folder, comitting, waiting for Fleet to apply, and removing the `crds` bundle folder, committing, and waiting for Fleet to apply.

File: docs/operator-manual/Rancher-Fleet.md (hunk 3 above)
Comment (viccuad) [reply docs/operator-manual/Rancher-Fleet.md]: Done in https://github.com/kubewarden/docs/pull/161/commits/b4321ddca87846628677ab41c40917f8b608bb21, suggestion was marked as outaded.

File: docs/operator-manual/Rancher-Fleet.md (hunk 5 above)
Comment (viccuad) [reply docs/operator-manual/Rancher-Fleet.md]: Done in https://github.com/kubewarden/docs/pull/161/commits/eae9a1cf5bf6643303c549ca2199971d47ae197c.

File: docs/operator-manual/Rancher-Fleet.md (hunk 4 above)
Comment (viccuad) [reply docs/operator-manual/Rancher-Fleet.md]: A better approach is to have the CRDs be in their own GitRepo, too. Done in https://github.com/kubewarden/docs/pull/161/commits/eae9a1cf5bf6643303c549ca2199971d47ae197c.

File: docs/operator-manual/Rancher-Fleet.md (hunk 2 above)
Comment (viccuad) [reply docs/operator-manual/Rancher-Fleet.md]: that sounds better, done in https://github.com/kubewarden/docs/pull/161/commits/eae9a1cf5bf6643303c549ca2199971d47ae197c.

**Review by flavio — APPROVED** [review]

**Comment by viccuad** [comment]
Moving to blocked, waiting for github.com/kubewarden/fleet-examples to be populated, and I can't push to it.

**Comment by flavio** [comment]
> Moving to blocked, waiting for https://github.com/kubewarden/fleet-example to be populated, and I can't push to it.

I should have fixed that, please try again

**Comment by viccuad** [comment]
Thanks, worked! Populated https://github.com/kubewarden/fleet-example, merging this PR.

**Pull request merged**